src/strands/multiagent/graph.py

```diff
@@ -384,6 +415,20 @@ def __init__(
         self.reset_on_revisit = reset_on_revisit
         self.state = GraphState()
         self.tracer = get_tracer()
+        self.session_manager = session_manager
+        self.hooks = hooks or HookRegistry()
+        if self.session_manager is not None:
+            self.hooks.add_hook(self.session_manager)
+
+        # Concurrent lock
+        self._lock = asyncio.Lock()
+        # Resume flag
+        self._resume_from_persisted = False
+        self._resume_next_nodes: list[GraphNode] = []
+
+        self._load_and_apply_persisted_state()
+        if not self._resume_from_persisted and self.state.status == Status.PENDING:
+            self.hooks.invoke_callbacks(MultiAgentInitializationEvent(orchestrator=self), supress_exceptions=True)
```

Comment:
This is tightly couple the session manager to hooks. I think the session_manager should be subscribing to `MultiAgentInitializationEvent` and performing the logic of 

```
self._load_and_apply_persisted_state()
        if not self._resume_from_persisted and self.state.status == Status.PENDING:
```

Agent and AgentInitialized follows a similar pattern: https://github.com/strands-agents/sdk-python/blob/08dc4aeaad6e75dc273d41a9898e0d08df09863b/src/strands/agent/agent.py#L350

But the hook should *not* be conditionally called; perhaps we need to provide more information to `MultiAgentInitializationEvent` to indicate whether or not it was being resumed?